src/docgen/view/documentation.ts

```diff
@@ -1,121 +1,227 @@
+import * as child from 'child_process';
+import * as os from 'os';
+import * as path from 'path';
+import * as util from 'util';
+import * as fs from 'fs-extra';
+import * as glob from 'glob-promise';
 import * as reflect from 'jsii-reflect';
+import { TargetLanguage } from 'jsii-rosetta';
+import { transliterateAssembly } from 'jsii-rosetta/lib/commands/transliterate';
 import { Markdown } from '../render/markdown';
 import { PythonTranspile } from '../transpile/python';
 import { Transpile } from '../transpile/transpile';
 import { TypeScriptTranspile } from '../transpile/typescript';
 import { ApiReference } from './api-reference';
 import { Readme } from './readme';
 
+const exec = util.promisify(child.exec);
+
 /**
- * Options for rendering documentation pages.
+ * Options for rendering a `Documentation` object.
  */
-export interface DocumentationOptions {
+export interface RenderOptions {
+
   /**
-   * The assembly to generate from.
+   * Include a generated api reference in the documentation.
+   *
+   * @default true
    */
-  readonly assembly: reflect.Assembly;
+  readonly apiReference?: boolean;
 
   /**
-   * Which language to generate docs for.
-   */
-  readonly language: string;
+    * Include the user defined README.md in the documentation.
+    *
+    * @default true
+    */
+  readonly readme?: boolean;
 
   /**
-   * Include a generated api reference in the documentation.
+    * Generate documentation only for a specific submodule.
+    *
+    * @default - Documentation is generated for the root module only.
+    */
+  readonly submodule?: string;
+
+}
+
+/**
+ * Options for creating a `Documentation` object.
+ */
+export interface DocumentationOptions {
+
+  /**
+   * Which language to generate docs for.
    *
-   * @default true
+   * @default 'ts'
    */
-  readonly apiReference?: boolean;
+  readonly language?: string;
```

Comment:
I'd be hard-pressed determining what are valid values to pass here... Should I pass `python` or `py` here? Is it `chsarp`, `C#`, `.NET`, or `dotnet`?

Either this needs to be a union, or an `enum`, or the documentation needs to tell me how I should name those...

Reply:
Changed to enum 👍

Reply:
@RomainMuller I ended up using an enum like class instead to add a transformation from a literal string to the object, and spare this switch casing from the caller.